### PR TITLE
235: Filter NodeJS dev dependencies

### PR DESCRIPTION
**Pull request by bsima**
I'm really not sure how to test this - I looked at "testing/quick" but couldn't figure out how to generate a `pkgMap` to run against a unit test. Any advice is appreciated.

Closes https://github.com/fossas/fossa-cli/issues/235

### CODE REVIEW

File: pkg/package.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ type Package struct {
 
 	Imports []Import
 	// SourceFiles []File
-	// Qualifier   Qualifier
+	Qualifier   Qualifier
```
Comment (bsima) [inline pkg/package.go]: Is this being actively detected, or do I have to write a function to detect Production vs Development?

Comment (liftM) [reply pkg/package.go]: I don't think these are implemented yet.

File: Makefile [hunk 2]
```diff
@@ -38,21 +38,21 @@ $(PREFIX)/fossa: $(BIN)/fossa
 
 # Building various Docker images.
 docker-base: ./docker/base/Dockerfile
-	sudo docker build -t quay.io/fossa/fossa-cli-base -f ./docker/base/Dockerfile .
+	docker build -t quay.io/fossa/fossa-cli-base -f ./docker/base/Dockerfile .
```
Comment (liftM) [inline Makefile]: These are `sudo` for Linux development. You might want to add MacOS-specific targets without `sudo`.

File: analyzers/nodejs/nodejs.go [hunk 3]
```diff
@@ -276,6 +276,15 @@ func recurseDeps(pkgMap map[pkg.ID]pkg.Package, p npm.Output) {
 		if ok {
 			continue
 		}
+
+        // Remove dev dependencies
+        if removeDev {
+            o := pkgMap[id]
+            if o.Qualifier == 2 {
```
Comment (liftM) [inline analyzers/nodejs/nodejs.go]: What is `o.Qualifier`?

File: Makefile (hunk 2 above)
Comment (bsima) [reply Makefile]: I do use Linux - sudo is only necessary if your user isn't part of the docker group, this is true of Linux and MacOS

    sudo groupadd docker
    sudo gpasswd -a $USER docker

c.f. https://askubuntu.com/questions/477551/how-can-i-use-docker-without-sudo/477554#477554

File: analyzers/nodejs/nodejs.go (hunk 3 above)
Comment (bsima) [reply analyzers/nodejs/nodejs.go]: This refers to the `Qualifier` field in the `Package` struct: https://github.com/fossas/fossa-cli/blob/49419e96778b772845535654cb0f813bb922b6d9/pkg/package.go#L60

File: Makefile (hunk 2 above)
Comment (liftM) [reply Makefile]: That's _true_, but I'm not sure that "being in the `docker` group" a good assumption.

1) Being in the `docker` group is effectively equivalent to being root, and I want to discourage this behaviour as much as possible.
2) You're effectively root anyway, so `sudo docker ...` is only more typing, not actually less secure.

**Comment by bsima** [comment]
Okay I’ll rebase that commit out in the morning

**Comment by bsima** [comment]
Crap, i rebased improperly and lost my commits... will fix this and re-open.

**Comment by bsima** [comment]
Superseded by https://github.com/fossas/fossa-cli/pull/270

**Pull request closed without merging**